index.js

```diff
@@ -1,287 +1,24 @@
-      }
-
-      if (reset) {
-        return resetEnvironment(environmentId, verified);
-      }
-    }, true)
-  );
-
-program
-  .command('doc')
-  .option('-s, --space-id <space-id>', 'Contentful space id')
-  .option('-e, --environment-id <environment-id>', 'Change the Contentful environment')
-  .option('-p, --path <path/to/docs>', 'Change the path where the docs are stored')
-  .option('-v, --verbose', 'Verbosity')
-  .option('-t, --template <path/to/template>', 'Use custom template for docs')
-  .option('--extension <file-extension>', 'Use custom file extension (default is `md`)')
-  .description('Generate offline docs from content-types')
-  .action(
-    actionRunner(async (cmd) => {
-      const config = await getConfig(parseArgs(cmd || {}));
-      const verified = await askMissing(config, ['accessToken', 'spaceId', 'environmentId']);
-      await createOfflineDocs(verified);
-    }, true)
-  );
-
-program
-  .command('content')
-  .requiredOption('--source-environment-id <environment-id>', 'Set the Contentful source environment (from)')
-  .requiredOption('--dest-environment-id <environment-id>', 'Set the Contentful destination environment (to)')
-  .option('-s, --space-id <space-id>', 'Contentful space id')
-  .option('-c, --content-type <content-type>', 'Specify content-type')
-  .option('-v, --verbose', 'Verbosity')
-  .option('-y, --yes', 'Assume "yes" as answer to all prompts and run non-interactively.')
-  .option('--diff', 'Manually choose skip/overwrite for every conflict')
-  .option('--force', 'No manual diffing. Overwrites all conflicting entries/assets')
-  .description('Transfer content from source environment to destination environment')
-  .action(
-    actionRunner(async (cmd) => {
-      const config = await getConfig(parseArgs(cmd || {}));
-      const verified = await askMissing({ ...config, environmentId: 'not-used' });
-
-      // run migrations on destination environment
-      await transferContent({
-        ...verified,
-        contentType: cmd.contentType || '',
-        forceOverwrite: cmd.force || false,
-        diffConflicts: cmd.diff || false,
-      });
-    })
-  );
-
-program.parse(process.argv);
+/**
+ * Adds utils for the migration
+ *
+ * Example:
+ * const withUtils = require('@jungvonmatt/contentful-migrations/utils');
+ *
+ * module.exports = withUtils(async (migration, context, utils) => {
+ *
+ *   ...
+ *
+ * });
+ *
+ */
+ const { getValidationHelper } = require('./lib/utils/validation');
+ const { getLocaleHelper } = require('./lib/utils/locale');
+ const { getMigrationHelper } = require('./lib/utils/migration');
+
+ // Export wrapper
+ module.exports.withUtils = (cb) => (migration, context) => {
+   const localeHelper = getLocaleHelper(migration, context);
+   const validationHelper = getValidationHelper(migration, context);
+   const migrationHelper = getMigrationHelper(migration, context);
+   return cb(migration, context, { ...localeHelper, ...validationHelper, ...migrationHelper });
```

Comment:
Die Namen der Util Funktion machen in ihrem Context Sinn.
Also "addInValues" im Context von validationHelper.
Aber nur als utils finde ich den Namen nicht aussagekräftig.
Warum nicht ein komplexeres Objekt zurückgeben, wo der Context als Name erhalten bleibt.
Dann kann man `utils.validation.addInValues` aufrufen.
Nima hat einige Hilfsfunktionen gebaut, die das Anlegen von Content erleichtern.
Das sollten wir dann da auch mit aufnehmen (gerne danach in einem extra PR, aber da würde dann auch ein extra Context "seeding" oder so Sinn machen).

Reply:
umgebaut